pkg/controller/start.go

```diff
@@ -102,29 +103,33 @@ func createManager(log logr.Logger, cfg config.Manager) (ctrl.Manager, error) {
 
 	opts := ctrl.Options{
 		Scheme:                 scheme,
-		Port:                   cfg.WebhookPort,
-		MetricsBindAddress:     cfg.MetricsAddr,
+		Metrics:                server.Options{BindAddress: cfg.MetricsAddr},
 		HealthProbeBindAddress: cfg.HealthProbeAddr,
 		LeaderElection:         cfg.EnableLeaderElection,
 		LeaderElectionID:       "hephaestus-controller-lock",
 	}
+	webhookOpts := webhook.Options{Port: cfg.WebhookPort}
 
 	if certDir := os.Getenv("WEBHOOK_SERVER_CERT_DIR"); certDir != "" {
 		log.Info("Overriding webhook server certificate directory", "value", certDir)
-		opts.WebhookServer = &webhook.Server{
-			CertDir: certDir,
-		}
+		webhookOpts.CertDir = certDir
 	}
 
+	restConfig := ctrl.GetConfigOrDie()
```

Comment:
Why are we setting this here now?

Reply:
sorry, it can move back. The old code did something weird and used the cacheFunc instead of the Cache options.
I will move it back since we won't be using cache.New().